types/Route.d.ts

```diff
@@ -0,0 +1,31 @@
+import { SvelteComponent, SvelteComponentTyped } from 'svelte'
+
+interface RouteProps {
+  path?: string
+  component?: SvelteComponent
+  [additionalProp: string]: unknown
+}
+
+interface RouteSlots {
+  default: {
+    location: RouteLocation
+    params: RouteParams
+  }
+}
+
+interface RouteLocation {
+  pathname: string
+  search: string
+  hash?: string
+  state: {
+    [k in string | number]: unknown
+  }
+}
+
+interface RouteParams {
+  [param: string]: string
+}
+
+type Route = SvelteComponentTyped<RouteProps, Record<string, any>, RouteSlots>
```

Comment:
```suggestion
const Route: SvelteComponentTyped<RouteProps, Record<string, any>, RouteSlots>
```

Reply:
Same as https://github.com/EmilTholin/svelte-routing/pull/117#discussion_r563926752

Reply:
Same as https://github.com/EmilTholin/svelte-routing/pull/117#discussion_r564248145, do you have any insight on this? at the moment, the only possibly viable solution that i could think of is to make use of the syntax proposed on an earlier comment of mine (https://github.com/EmilTholin/svelte-routing/pull/117#issuecomment-754306957), in short, encapsulating every `.d.ts` into a `declare` modifier might solve the problem cited

What's your view on that?

Reply:
Solved by https://github.com/EmilTholin/svelte-routing/pull/117#discussion_r564284217